### PR TITLE
#8834: Use filtered admin URLs in do_robots()

### PR DESCRIPTION
**Pull request by masteradhoc**
Updates `do_robots()` to use filtered admin URL paths instead of hardcoded `/wp-admin/` values.

This changes the default `robots.txt` output to derive the `Disallow` and `Allow` paths from `admin_url()` and `admin_url( 'admin-ajax.php' )`, while continuing to output only the path portion. This allows installs that filter admin URLs to generate correct `robots.txt` directives.

A PHPUnit regression test is included to verify that filtered admin paths are reflected in the `do_robots()` output.

Trac ticket: https://core.trac.wordpress.org/ticket/63467

## Use of AI Tools
AI assistance: Yes  
Tool(s): GitHub Copilot  
Model(s): GPT-5.4  
Used for: Drafting the initial patch and regression test. The final implementation, test coverage, and PR description were reviewed and edited by me.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/functions.php [hunk 1]
```diff
@@ -1718,10 +1720,8 @@ function do_robots() {
 	$output = "User-agent: *\n";
 	$public = (bool) get_option( 'blog_public' );
 
-	$site_url = parse_url( site_url() );
-	$path     = ( ! empty( $site_url['path'] ) ) ? $site_url['path'] : '';
-	$output  .= "Disallow: $path/wp-admin/\n";
-	$output  .= "Allow: $path/wp-admin/admin-ajax.php\n";
+	$output .= 'Disallow: ' . (string) wp_parse_url( admin_url(), PHP_URL_PATH ) . "\n";
```
Comment (westonruter) [inline src/wp-includes/functions.php:1728]: ✅ 

```
wp> (string) wp_parse_url( admin_url(), PHP_URL_PATH );
=> string(10) "/wp-admin/"
```

File: src/wp-includes/functions.php [hunk 2]
```diff
@@ -1718,10 +1720,8 @@ function do_robots() {
...
+	$output .= 'Allow: ' . (string) wp_parse_url( admin_url( 'admin-ajax.php' ), PHP_URL_PATH ) . "\n";
```
Comment (westonruter) [inline src/wp-includes/functions.php:1729]: ✅ 

```
wp> (string) wp_parse_url( admin_url( 'admin-ajax.php' ), PHP_URL_PATH )
=> string(24) "/wp-admin/admin-ajax.php"
```

**Review by westonruter — APPROVED** [review]

File: src/wp-includes/functions.php [hunk 3]
```diff
@@ -1711,7 +1711,9 @@ function do_feed_atom( $for_comments ) {
  *              filter callback.
  */
 function do_robots() {
-	header( 'Content-Type: text/plain; charset=utf-8' );
+	if ( ! headers_sent() ) {
+		header( 'Content-Type: text/plain; charset=utf-8' );
+	}
```
Comment (mukeshpanchal27) [inline src/wp-includes/functions.php:1716]: Why was this change needed?

Did we encounter an issue that this fixes, or is this something suggested by AI rather than being based on an actual problem?

File: src/wp-includes/functions.php (hunk 1 above)
Comment (mukeshpanchal27) [reply src/wp-includes/functions.php:1728]: What about below @westonruter?

```
$admin_path = wp_parse_url( admin_url(), PHP_URL_PATH );

$output .= "Disallow: {$admin_path}\n";
$output .= "Allow: {$admin_path}admin-ajax.php\n";
```

File: src/wp-includes/functions.php (hunk 3 above)
Comment (masteradhoc) [reply src/wp-includes/functions.php:1716]: @mukeshpanchal27 while i was testing i received the issue that header was already sent. Therefore i added this safeguard. You shouldnt have encountered it while testing now, right?

File: src/wp-includes/functions.php (hunk 1 above)
Comment (westonruter) [reply src/wp-includes/functions.php:1728]: It's better if we call `admin_url()` multiple times so that the filter applies twice so the URLs can be customized.

File: src/wp-includes/functions.php (hunk 3 above)
Comment (westonruter) [reply src/wp-includes/functions.php:1716]: It makes sense for me to add. This is done many places in core, including in `status_header()`.

Comment (westonruter) [reply src/wp-includes/functions.php:1716]: That said, the pattern used throughout core may be a bad idea because it masks cases where the header is not getting set as expected.

File: src/wp-includes/functions.php [hunk 4]
```diff
@@ -1723,10 +1725,8 @@ function do_robots() {
 	$output = "User-agent: *\n";
 	$public = (bool) get_option( 'blog_public' );
 
-	$site_url = parse_url( site_url() );
-	$path     = ( ! empty( $site_url['path'] ) ) ? $site_url['path'] : '';
-	$output  .= "Disallow: $path/wp-admin/\n";
-	$output  .= "Allow: $path/wp-admin/admin-ajax.php\n";
+	$output .= 'Disallow: ' . (string) wp_parse_url( admin_url(), PHP_URL_PATH ) . "\n";
+	$output .= 'Allow: ' . (string) wp_parse_url( admin_url( 'admin-ajax.php' ), PHP_URL_PATH ) . "\n";
```
Comment (westonruter) [inline src/wp-includes/functions.php:1729]: FYI: I'm removing the redundant `(string)` type casts. These are unnecessary after conditional return types were added in https://github.com/WordPress/wordpress-develop/commit/0fced49b8dbc4ab35b3b781bd38937ab938e54f4.
